dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/logging/LogUtils.java

```diff
@@ -264,36 +264,64 @@ protected static Logger createLogger(final Class<?> cls,
                 try {
                     b = BundleUtils.getBundle(cls, bundleName);
                 } catch (MissingResourceException rex) {
-                    //ignore
                 }
             }
             if (b != null) {
                 b.getLocale();
             }
 
             if (loggerClass != null) {
-                try {
-                    Constructor<?> cns = loggerClass.getConstructor(String.class, String.class);
-                    if (name == null) {
+                // Default logger of Liberty is WsLogger. 
+                // CXFLogger is inherited from this WsLogger class to preserve same behavior
+                // except the one that we aimed to change. Which is to add class name to log records
+                if (loggerClass.equals(WsLogger.class)) {
+                    try {
+                        Constructor<?> cns = loggerClass.getConstructor(String.class, String.class, Class.class); //Liberty change: Class is added as parameter
+                        if (name == null) {
```

Comment:
What is the point of this check? 

This try catch will either return a new instance of `CXFLogger` or throw an ite, but no where in the `if` statement is it referenced, and the following try catch after the `if` statement is identical. If `name` isn't use, could you just drop the `if` block and keep a single try catch?